### PR TITLE
docs: add codeowners, backend-api-requirements

### PR DESCRIPTION
**Pull request by pedromsmoreira**
- Add CODEOWNERS
- Add backend-api requirements
- Fix typo

### CODE REVIEW

File: backend-api/requirements.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - Check in/out Pets
+- Publish Pets for adoption
+- Manage owners
+- Blacklist/whitelist owners
+
+## Implementation Ideas
+
+Since these entities have a lot of connections, I would like to use a graph database.
+
+Options:
```
Comment (pedromsmoreira) [inline backend-api/requirements.md:17]: @joao-r-reis You can add more techs if you know more of them.

Comment (joao-r-reis) [reply backend-api/requirements.md:17]: [JanusGraph](https://janusgraph.org/), [Azure Cosmos DB](https://docs.microsoft.com/en-us/azure/cosmos-db/graph-introduction), [Amazon Neptune](https://aws.amazon.com/neptune/)

**Review by joao-r-reis — APPROVED** [review]
:+1:

File: backend-api/requirements.md (hunk 1 above)
Comment (pedromsmoreira) [reply backend-api/requirements.md:17]: I will add those. That way they stay as an option

**Pull request merged**